caffe2/CMakeLists.txt

```diff
@@ -1365,6 +1368,16 @@ if(USE_DISTRIBUTED)
       target_compile_definitions(torch_cuda PUBLIC USE_C10D_NCCL)
     endif()
   endif()
+  if(USE_C10D_XCCL)
+    # if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    #   set_source_files_properties(
+    #     ${TORCH_SRC_DIR}/csrc/distributed/c10d/ProcessGroupXCCL.cpp
+    #     PROPERTIES COMPILE_DEFINITIONS  "CCL_ENABLE_SYCL")
+    #   target_sources(torch_xpu PRIVATE ${TORCH_SRC_DIR}/csrc/distributed/c10d/ProcessGroupXCCL.cpp)
+    # endif()
+    target_compile_definitions(torch_xpu PUBLIC USE_C10D_XCCL)
+    target_compile_definitions(torch_xpu PUBLIC CCL_ENABLE_SYCL)
```

Comment:
The `CCL_ENABLE_SYCL` flag is added to support C++ compilation of oneCCL-related code. Directly adding `COMPILE_DEFINITIONS` to `ProcessGroupXCCL.cpp` does not seem to work (as seen in the commented-out code). This is a workaround, and discussions are ongoing with oneCCL regarding compiler selection issues.

Reply:
CCL_ENABLE_SYCL should be always on for GPU?
Can we remove it directly?